### PR TITLE
feat(connector): add `refineLater`

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

**Result**

### CODE REVIEW

File: src/connectors/configure/connectConfigure.ts [hunk 1]
```diff
@@ -116,6 +133,7 @@ const connectConfigure: ConfigureConnector = function connectConfigure(
         renderFn(
           {
             refine: connectorState.refine!,
+            setSearchParametersWithoutSearch: connectorState.setSearchParametersWithoutSearch!,
```
Comment (Haroenv) [inline src/connectors/configure/connectConfigure.ts:136]: `refineWithoutApplying`?

Comment (eunjae-lee) [reply src/connectors/configure/connectConfigure.ts:136]: It's not refining at the moment, so `refineLater` sounds more appropriate than `refineWithoutApplying`. And `draft` sounds like there should be `commit` or `save`, etc.
I struggled the name 😂 and so far I love `refineLater`.

Comment (francoischalifour) [reply src/connectors/configure/connectConfigure.ts]: What about something like `batchRefine`?

`refineLater` doesn't give a clue as to when it's going to refine.

Comment (eunjae-lee) [reply src/connectors/configure/connectConfigure.ts]: IMO, `batch` feels like there should be a dedicated trigger for that.
Or, what if we use terms like `schedule` or `defer`?
As in `scheduleRender`, ...

**Comment by eunjae-lee** [comment]
I created this branch as a stepping stone for https://github.com/algolia/instantsearch.js/pull/4446
and I guess we can use helper directly there.
So I'll close this. If we need this functionality somewhere else, then we can bring this back.

**Pull request closed without merging**